値条件式の単一条件規則で rustc の型エラー

以下のコードで rustc のエラーが発生する

```shol
*a
.1#2
```

```
[*] Compiling...
error[E0308]: mismatched types
  --> example/fizzbuzz.rs:51:27
   |
51 |           if v.clone() == 1 {
   |              ---------    ^ expected `bool`, found integer
   |              |
   |              expected because this is `bool`

error[E0277]: can't compare `String` with `{integer}`
  --> example/fizzbuzz.rs:57:24
   |
57 |           if v.clone() == 1 {
   |                        ^^ no implementation for `String == {integer}`
   |
   = help: the trait `PartialEq<{integer}>` is not implemented for `String`
   = help: the following other types implement trait `PartialEq<Rhs>`:
             `String` implements `PartialEq<&str>`
             `String` implements `PartialEq<Cow<'_, str>>`
             `String` implements `PartialEq<str>`
             `String` implements `PartialEq`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
rustc returned with non-zero exit code
```

--- a/src/code_generator.rs
+++ b/src/code_generator.rs
@@ -664,21 +664,24 @@ fn generate_single_condition_rule(
     let cond_meta = cond.meta.as_ref().unwrap();
 
     // if condition { 部分を生成
-    write!(f, "          ")?;
     match cond_meta.kind {
         // 条件式と一致したら出力式を push
         ConditionKind::Equal(_) => {
-            write!(f, "if {}.clone() == ", Identf::V_VALUE_REF)?;
-            generate_expr(f, &cond.expr, &|_, _| {
-                unreachable!()
-            })?;
-            write!(f, " ")?;
+            let (result_type, expr_str) = {
+                let mut buffer = Vec::new();
+                let result_type = generate_expr(&mut buffer, &cond.expr, &|_, _| {
+                    unreachable!()
+                })?;
+                (result_type, String::from_utf8(buffer).unwrap())
+            };
+            if result_type != capture_type { return Ok(()); }
+            write!(f, "          if {}.clone() == {} ", Identf::V_VALUE_REF, expr_str)?;
         },
         // 無条件で出力式を push
         ConditionKind::Capture(_) => (),
         // 条件式を満たすならば出力式を push
         ConditionKind::CaptureCondition(_) => {
-            write!(f, "if ")?;
+            write!(f, "          if ")?;
             generate_expr(f, &cond.expr, &|f, _| {
                 write!(f, "{}.clone()", Identf::V_VALUE_REF)?;
                 Ok(capture_type)

